gapic/src/main/com/google/gapid/perfetto/views/FrameEventsSummaryPanel.java

```diff
@@ -403,6 +408,16 @@ public void computeSelection(CombiningBuilder builder, Area area, TimeSpan ts) {
     }
   }
 
+  private static FrameSelection getSelectedFrames(State state) {
+    Selection selection = state.getSelection(Selection.Kind.FrameEvents);
+    if (selection instanceof FrameEventsTrack.Slice) {
+      return ((FrameEventsTrack.Slice) selection).getSelection();
+    } else if (selection instanceof FrameEventsTrack.Slices) {
```

Comment:
Can you combine FrameEventsTrack.Slice with FrameEventsTrack.Slices?  Stella did something similar.

Reply:
I'll look into what Stella did and make changes if its suitable here.

Reply:
To avoid type checking here, it will require `FrameEventsTrack.Slice` and `FrameEventsTrack.Slices` both implement `Selection` framework, and have them both saved under a new category `Kind.FurtherFrame`.
I can see the difficulty of letting them implementing Selection framework here, because the class's contains signature is `contains(long[] f, String[] l)` rather than `contains(long id)`, if we really can't find a long-typed id as a key, I'm fine with it checked in as this for now.
There are now two outliners doing type checking, one her and one in VulkanEvent track. We'd better think about how to add some flexibility into the selection framework to make these two parts more elegant.

Reply:
As Stella pointed out, grouping them together is going to involve a lot of changes to the Selection framework and I'd like to avoid mixing those things up in the change. Stella and I will look into refactoring the Selection framework further, if we stumble upon this usecase again. Right now, only the Vulkan events and Frame events require this behavior.